### PR TITLE
core: resumable overlay transition runs

### PR DESCRIPTION
**Pull request by jsign**
This PR saves to the db the start and end markers for the overlay transition. 
This is done to detect on a re-run if the overlay transition has finished. 

It can't still resume an in-progress (interrupted) transition.

### CODE REVIEW

File: core/state/database.go
Comment (jsign) [inline core/state/database.go:148]: Adding a `error` return in `NewDatabaseWithConfig` seems like a big step since it has an impact on many call sites.

Not that's strictly a problem, but I'd like your input on this matter before some "exploding" change stream. I'm not convinced of this panic, but leaving it now just to save our time while doing benchmarks temporarily.

If we want to avoid adding an error parameter in this method, feels like the start/end markers should be passed by callers (e.g: in existing parameters, ne

**Review by gballet — APPROVED** [review]
That's going to work for the replay, but it's not enough for interrupting geth during the hard fork: if there is a reorg or something, or simply that the latest saved full block is before the transition, this will cause nodes to mark the transition as complete when it's not upon resuming.

It's good enough for now, though. I'll merge when it's confirmed to work on replay.

**Comment by jsign** [comment]
> That's going to work for the replay, but it's not enough for interrupting geth during the hard fork: if there is a reorg or something, or simply that the latest saved full block is before the transition, this will cause nodes to mark the transition as complete when it's not upon resuming.
> 
> It's good enough for now, though. I'll merge when it's confirmed to work on replay.

This seems to have worked fine. After the restart, I confirmed the markers were read correctly from the database.

As we talked out of band, there might be more things that might worth saving but apparently aren't needed.

I'll leave it up to you to merge it or not.

**Comment by jsign** [comment]
Superseded by https://github.com/gballet/go-ethereum/pull/223

**Pull request closed without merging**
